File: 2022/P01/problem1a.py
data = []
with open("problem1.txt", 'r') as f:
    data = f.read().strip('\n').split('\n')

data_s = ' '.join(data)
unfolded_arr = [[int(x) for x in s.split(' ')] for s in data_s.split('  ')]
sum_arr = [sum(a) for a in unfolded_arr]
print(max(sum_arr))

#data = [int(x.strip()) if x != '\n' else x.strip() for x in data]
#cur_best = 0
#cur_total = 0
#for d in data:
#    if d == '':
#        cur_best = max(cur_best, cur_total)
#        cur_total = 0
#    else:
#        cur_total += d
#print(cur_best)

File: 2022/P01/problem1a.rs
use std::fs;

fn main() {
    let data = fs::read_to_string("problem1.txt")
        .expect("Failed to read file");
    let data = data.trim()
        .split('\n');
    let mut max = 0;
    let mut accum = 0;
    for el in data {
        match el.trim().parse::<u32>() {
            Ok(num) => {
                accum += num;
            },
            Err(_) => {
                max = if accum > max { accum } else { max };
                accum = 0;
                continue;
            },
        }
    }
    println!("{max}");
}
